lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2585,6 +2703,34 @@ LLVMToSPIRV::transLinkageType(const GlobalValue *GV) {
   return SPIRVLinkageTypeKind::LinkageTypeExport;
 }
 
+LLVMToSPIRV::FPContract LLVMToSPIRV::getFPContract(Function *F) {
+  auto It = FPContractMap.find(F);
+  if (It == FPContractMap.end()) {
+    return FPContract::UNDEF;
+  }
+  return It->second;
```

Comment:
Honestly, I don't know what is default value for enumerations, but theoretically, the whole function body could be replaced by a call to [lookup method](https://llvm.org/doxygen/classllvm_1_1DenseMapBase.html#a0b2ca98dc28c61793ff5c90d23e5f14e)

Reply:
Default value should be zero (matches `FPContract::UNDEF`), so `lookup` can be used here.
Current code is more explicit though, so I prefer to keep it unless you suggest to do otherwise.